### PR TITLE
Added variable number for tracking

### PR DESCRIPTION
**Pull request by edkazcarlson**
Added variable number to NativeVarInfo
so 3rd party applications can track variables easier
with the help of Andrew Au.
@cshung

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=39861) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=39861) before we can accept your contribution.<br/><br/>:x: edkazcarlson [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=39861)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=39861) it.</sub>

File: src/coreclr/src/tools/aot/ILCompiler.Reflection.ReadyToRun/x86/CallPattern.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Licensed to the .NET Foundation under one or more agreements.
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (mchandschuh) [inline src/coreclr/src/tools/aot/ILCompiler.Reflection.ReadyToRun/x86/CallPattern.cs:1]: This change appears to be unintentional.

Comment (edkazcarlson) [reply src/coreclr/src/tools/aot/ILCompiler.Reflection.ReadyToRun/x86/CallPattern.cs]: Fixed, thank you.

File: src/coreclr/src/tools/aot/ILCompiler.Reflection.ReadyToRun/DebugInfo.cs [hunk 2]
```diff
@@ -68,7 +68,6 @@ public static string GetPlatformSpecificRegister(Machine machine, int regnum)
                 case Machine.Amd64:
                     return ((Amd64.Registers)regnum).ToString();
                 case Machine.Arm:
-                case Machine.ArmThumb2:
```
Comment (cshung) [inline src/coreclr/src/tools/aot/ILCompiler.Reflection.ReadyToRun/DebugInfo.cs:71]: It appears that the script is overwriting some expected changes like this one. We need to make sure we only add the expected changes.

File: src/coreclr/src/tools/aot/ILCompiler.Reflection.ReadyToRun/ReadyToRunSignature.cs [hunk 3]
```diff
@@ -1254,7 +1254,6 @@ private ReadyToRunSignature ParseSignature(ReadyToRunFixupKind fixupType, String
 
 
                 case ReadyToRunFixupKind.Check_TypeLayout:
-                case ReadyToRunFixupKind.Verify_TypeLayout:
```
Comment (cshung) [inline src/coreclr/src/tools/aot/ILCompiler.Reflection.ReadyToRun/ReadyToRunSignature.cs:1257]: Same here with all the TypeLayout related changes.

**Review by cshung — CHANGES_REQUESTED** [review]

**Review by cshung — APPROVED** [review]
Looks good to me

**Pull request merged**